### PR TITLE
Setup BLEmulator

### PR DESCRIPTION
**Pull request by mikolak**

### CODE REVIEW

File: android/src/main/java/com/polidea/blemulator/SimulatedAdapter.java [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+        return null;
+    }
+
+    @Override
+    public void startDeviceScan(String[] filteredUUIDs,
+                                int scanMode,
+                                int callbackType,
+                                OnEventCallback<ScanResult> onEventCallback,
+                                OnErrorCallback onErrorCallback) {
+        Log.w("SIMULATOR", "START DEVICE SCAN");
```
Comment (pawelByszewski) [inline android/src/main/java/com/polidea/blemulator/SimulatedAdapter.java]: Let's add `log.i` everywhere

File: android/src/main/java/com/polidea/blemulator/bridging/DartMethodCaller.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import io.flutter.plugin.common.MethodChannel;
+
+public class DartMethodCaller {
+    private MethodChannel dartMethodChannel;
+
+    public DartMethodCaller(MethodChannel dartMethodChannel) {
+        this.dartMethodChannel = dartMethodChannel;
+    }
+
```
Comment (bartoszwilk) [inline android/src/main/java/com/polidea/blemulator/bridging/DartMethodCaller.java]: Unnecessary empty lines

File: android/src/main/java/com/polidea/blemulator/bridging/constants/ChannelName.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.polidea.blemulator.bridging.constants;
+
+public interface ChannelName {
+    String BASE = "com.polidea.blemulator";
+
+    String TO_DART = BASE + "/to_dart";
```
Comment (bartoszwilk) [inline android/src/main/java/com/polidea/blemulator/bridging/constants/ChannelName.java]: We use pascal case for all bridge-related names if I recall correctly.

File: lib/blemulator/blemulator_core.dart [hunk 4]
```diff
@@ -0,0 +1,12 @@
+part of blemulator;
+
+class Blemulator {
+
+  BlemulatorBridge _bridge;
+
+  Blemulator() {
+    _bridge = BlemulatorBridge();
```
Comment (bartoszwilk) [inline lib/blemulator/blemulator_core.dart]: Why don't we just instantiate this bridge in it's declaration line? We don't inject it (maybe we should?) so currently there is no need to have this default constructor.

**Pull request merged**